src/commands/hld/reconcile.ts

```diff
@@ -101,6 +102,39 @@ type MiddlewareMap<T = Partial<ReturnType<typeof middleware.create>>> = {
   default?: T;
 };
 
+// In spk hld reconcile, the results should always result in the same artifacts being created based on the state of bedrock.yaml.
+// The only exception is for files under the /config directories and any access.yaml files.
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const purgeRepositoryComponents = (
+  absHldPath: string,
+  repositoryName: string
+): void => {
+  assertIsStringWithContent(absHldPath, "hld-path");
+  assertIsStringWithContent(repositoryName, "repository-name");
+
+  const filesToDelete = getAllFilesInDirectory(
+    path.join(absHldPath, repositoryName)
+  ).filter(
+    (filePath) =>
+      !filePath.match(/access\.yaml$/) && !filePath.match(/config\/.*\.yaml$/)
```

Comment:
just a note. regular expression uses more resource (CPU), we try to avoid it if we can. e.g. `filePath.match(/access\.yaml$/)` can be `file.endsWith(access.yaml)